### PR TITLE
Move authN/Z code to separate module

### PR DESCRIPTION
**Pull request by snazy**
Also adds some unit tests

### CODE REVIEW

File: servers/quarkus-auth/src/main/java/org/projectnessie/server/authn/NessieHttpAuthenticator.java [hunk 1]
```diff
@@ -61,40 +60,56 @@ public NessieHttpAuthenticator(
...
-                }
+            securityIdentity -> base.maybeTransform(securityIdentity, context.request().path()));
+  }
 
-                // Disallow unauthenticated requests when requested by configuration.
-                // Note: Quarkus by default permits unauthenticated requests unless there are
-                // specific authorization rules that validate the security identity.
-                throw new AuthenticationFailedException("Missing or unrecognized credentials");
-              }
+  static class BaseNessieHttpAuthenticator {
```
Comment (snazy) [inline servers/quarkus-auth/src/main/java/org/projectnessie/server/authn/NessieHttpAuthenticator.java:81]: Introduced this indirection for tests

File: servers/quarkus-auth/src/main/java/org/projectnessie/server/authz/CompiledAuthorizationRules.java [hunk 2]
```diff
@@ -62,7 +62,7 @@ private Map<String, Script> compileAuthorizationRules() {
     rules.forEach(
         (key, value) ->
             scripts.computeIfAbsent(
-                value,
+                key,
```
Comment (snazy) [inline servers/quarkus-auth/src/main/java/org/projectnessie/server/authz/CompiledAuthorizationRules.java:65]: This was actually a bug :)

File: servers/quarkus-auth/src/test/java/org/projectnessie/server/authn/TestNessieHttpAuthenticator.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      QuarkusSecurityIdentity.builder().setPrincipal(() -> "some-user").build();
+
+  @Test
+  public void authentication() {
+    NessieHttpAuthenticator.BaseNessieHttpAuthenticator baseAuth =
+        new NessieHttpAuthenticator.BaseNessieHttpAuthenticator(
+            new QuarkusNessieAuthenticationConfig() {
+              @Override
+              public boolean enabled() {
+                return false; // not evaluated
```
Comment (dimas-b) [inline servers/quarkus-auth/src/test/java/org/projectnessie/server/authn/TestNessieHttpAuthenticator.java:49]: nit: maybe `UnsupportedOperationException` instead?

**Review by dimas-b — APPROVED** [review]

File: servers/quarkus-auth/src/test/java/org/projectnessie/server/authn/TestNessieHttpAuthenticator.java (hunk 3 above)
Comment (dimas-b) [reply servers/quarkus-auth/src/test/java/org/projectnessie/server/authn/TestNessieHttpAuthenticator.java:49]: Actaully, if it's not used in test call paths, why not keep it inside `NessieHttpAuthenticator`?

**Pull request merged**